File: MediCareWebSpa/src/app/DTOs/models/service.ts
import { Speciality } from "./speciality";

export class Service {
  id: number;
  name: string;
  description: string;
  price: number;
  durationMinutes: number;
  specialityId: number;
  speciality: Speciality;

  constructor(
    id: number,
    name: string,
    description: string,
    price: number,
    durationMinutes: number,
    specialityId: number,
    speciality: Speciality
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
    this.durationMinutes = durationMinutes;
    this.specialityId = specialityId;
    this.speciality = speciality;
  }
}

File: MediCareWebApi/Models/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class Service : Entity
	{
		[Required]
		[MaxLength(50)]
		public string Name { get; set; }

		[Required]
		[MaxLength(256)]
		public string Description { get; set; }

		[Required]
		public decimal Price { get; set; }

		[Required]
		public int DurationMinutes { get; set; }

		[Required]
		public int SpecialityId { get; set; }

		[ForeignKey("SpecialityId")]
		public Speciality Speciality { get; set; }
	}
}
